### DISCUSSION TITLE
Using Refine with Prisma and next-auth

### DISCUSSION THREAD

**Discussion by cloudcompute** [refinedev/refine#1812] [Q&A] [answered]
Hi

This package certainly looks promising. Pl. consider this use case.. A company is looking for a ```single User Interface panel``` for all its requirements. Its various users

- ```employees``` from several departments
- ```content writers```
- ```B2B partners```

should have ```RBAC/ABAC```

- view only those UI screens/components and 
- perform only those actions (Create, Update, Delete) that they are supposed to perform

My questions:

a. If I want to write this single UI panel using Next.Js, I need to use the Refine's Next.js router. Am I right?

b. To save the data entered using the admin panel, use the Refine's db provider. Do I need to write a DB provider of my own if I want to use ```Prisma```, or there already exists one for it? I think one has to bypass the Refine's DB provider and access the Prisma API directly.

b. When I need to pull the content i.e. images and rich text and feed it to a ```SSG``` (static-site-generator) application written in Next.js, Refine won't come in picture.. and I will retrieve the content using the backend API directly. Correct?

d. What do you mean when you said this:

```
React is strongly opinionated about three parts of your application:
- API Networking
- State Management
- Authentication & Authorization
```

Does this mean one cannot replace ```react-query``` with any other state-management technique?

What limitations does Refine place for API networking and AuthN / AuthZ? Won't it allow me to use [next-auth](https://github.com/nextauthjs/next-auth) like library for the UI panel that I want to write using Nextjs?

Thank you in advance.

**Comment by cloudcompute** [comment]
I am reading the tutorials. Looks like, Refine does not seem to fit well in a scenario where someone wants to use Prisma and next-auth.

**Comment by omeraplak** [comment, marked as answer]
Hey @asynchor 👋,
Thank you for contacting us! 

a. Yes, You should use the `@pankod/refine-nextjs-router` package in Next.js projects
b. You can use it by adding a REST or GraphQL API layer in front of Prisma (I found a Prisma provider written for `react-admin`, maybe it can give you an idea, https://github.com/codeledge/ra-data-simple-prisma)
c. You can use it as it is done in Refine's finefoods client. You can prefer SSG over SSR, https://github.com/pankod/refine/blob/master/examples/fineFoods/client/pages/index.tsx#L109 
d. It is not possible to change the state management. The only way would be to move forward with a fork. I think you can use `next-auth` with refine but we haven't tried it. If you have a problem we can help

I hope it will be useful for you 🚀

**Reply by doender** [reply]
@cloudcompute Did you every manage to create a prisma provider based on react-admin? If so, I would be very interested :) If not, I'll think about creating it myself

**Comment by cloudcompute** [comment]
Hi @omeraplak 

Thanks for the clarification. 

react-query is the best. I just asked about it to clarify the point stated in the documentation .. ```Refine strongly opinionated about three parts``` and one is state management.

I will have a look at the links you have shared.

**Comment by crubier** [comment]
Would be interested in use with Next-Auth

**Reply by vinclou** [reply]
Here's an example https://refine.dev/blog/nextauth-google-github-authentication-nextjs/
